File: II-semester/Lista 1/LiczbyPierwsze: Java/LiczbyPierwsze.java
import java.util.Arrays;

public class LiczbyPierwsze 
{
    public static boolean tablica[];

    public LiczbyPierwsze (int n)
    {
     	tablica = new boolean[n + 1];
     	Arrays.fill(tablica, true); //wypełnienie tablicy wartościami true

        tablica[0] = false;
        tablica[1] = false;

        for (int i = 2; i * i <= n; i++)
        {
            if (tablica[i])
            {
                for (int j = i + i; j <= n; j = j + i)
                {
                    tablica[j] = false;
                }
            }
        }
    }

    public static int liczba (int m)
    {
        int ktoraPierwsza = 0; //m-ta liczba pierwsza w tablicy
        int mtaPierwsza = 0;
        for (int i = 2; i < tablica.length; i++)
        {
            if (tablica[i])
            {
                mtaPierwsza = i;
                if (ktoraPierwsza == m) //znaleźliśmy m-tą liczbę pierwszą
                {
                    break;
                }
                ktoraPierwsza++;
            }
        }
        return mtaPierwsza;
    }
}

File: II-semester/Lista 1/LiczbyPierwsze: C++/src/LiczbyPierwsze.cpp
#include <iostream>
#include "header.hpp"

LiczbyPierwsze::LiczbyPierwsze(int n)
{
    for (int i = 0; i <= n + 1; i++)
    {
        tablica.push_back(true);
    }
    
    tablica[0] = false;
    tablica[1] = false;
    
    for (unsigned int i = 2; i * i <= unsigned(n + 1); i++)
    {
        if (tablica[i])
        {
            for (unsigned int j = i + i; j <= unsigned(n + 1); j = j + i)
            {
                tablica[j] = false;
            }
        }
    }
}


unsigned int LiczbyPierwsze::liczba(int m)
{
    {
    	int ktoraPierwsza = 0; //m-ta liczba pierwsza w tablicy
    	unsigned int mtaPierwsza = 0;
    	for (unsigned int i = 2; i < tablica.size(); i++)
        {
  	    if (tablica[i])
    	    {
    	        mtaPierwsza = i;
                if (ktoraPierwsza == m) //znaleźliśmy m-tą liczbę pierwszą
                {
                    break;
                }
                ktoraPierwsza++;
    	    }
    	}
    	return mtaPierwsza;
    }
}

